kyuubi-common/src/main/scala/org/apache/kyuubi/Utils.scala

```diff
@@ -137,12 +137,18 @@ object Utils extends Logging {
   /**
    * Delete a directory recursively.
    */
-  def deleteDirectoryRecursively(f: File): Boolean = {
-    if (f.isDirectory) f.listFiles match {
-      case files: Array[File] => files.foreach(deleteDirectoryRecursively)
-      case _ =>
+  def deleteDirectoryRecursively(f: File, forceDelete: Boolean = true): Unit = {
+    if (f.isDirectory) {
+      val files = f.listFiles
+      if (files != null && files.nonEmpty) {
+        files.foreach(deleteDirectoryRecursively(_, forceDelete))
+      }
+    }
+    try {
+      f.delete()
+    } catch {
+      case e: Exception if !forceDelete => throw e
```

Comment:
I think you can rename `forceDelete` to `ignoreException`, or others, due it does not force delete the file if exception thrown.

And we can log exception error here.

Reply:
No problem. `ignoreException` was the original param name in my first draft, changing back to it.  Fixed.